### PR TITLE
Hackathon frontend and backend changes

### PR DESCRIPTION
**Pull request by adamabeshouse**
1) Adjust web API to return SAMPLE id lists instead of patient id lists (confusion due to table being called patient_list :stuck_out_tongue: )

2) Cache data by sample list as well as by sample, so that, when possible, we can access by sample list instead of querying a ton of different indexes for samples.

3) Hackathon performance improvements to frontend

### CODE REVIEW

**Comment by jjgao** [comment]
@pieterlukasse could you review this pr?

**Comment by jjgao** [comment]
@adamabeshouse could you please resolve the conflicts?

File: portal/src/main/webapp/js/api/cbioportal-datamanager.js [hunk 1]
```diff
@@ -654,14 +656,20 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
...
 								def.reject();
 							}).then(function() {
-								return window.cbioportal_client.getGeneticProfileData({genetic_profile_ids: dm_ret.getGeneticProfileIds(), genes: dm_ret.getQueryGenes(), sample_ids: dm_ret.getSampleIds()});
+                                                            if (case_set_properties.case_set_id === "-1") {
+                                                                // custom case list
+                                                                console.log("by samples");
+								return window.cbioportal_client.getGeneticProfileDataBySample({genetic_profile_ids: dm_ret.getGeneticProfileIds(), genes: dm_ret.getQueryGenes(), sample_ids: dm_ret.getSampleIds()});
+                                                            } else {
+                                                                console.log("by sample list");
+                                                                return window.cbioportal_client.getGeneticProfileDataBySampleList({genetic_profile_ids: dm_ret.getGeneticProfileIds(), genes: dm_ret.getQueryGenes(), sample_list_id: [case_set_properties.case_set_id]});
```
Comment (pieterlukasse) [inline portal/src/main/webapp/js/api/cbioportal-datamanager.js]: maybe we should decide whether to call it case_set_id in the backend as well?

**Comment by adamabeshouse** [comment]
@pieterlukasse thank you for your review. Please see my responses. Let me know what you think.

**Comment by pieterlukasse** [comment]
@adamabeshouse : I think at least the following items could be fixed:
- ProfileDataMapper.xml SQL duplication
- Add extra comments/documentation in the code of ApiService.java (as indicated in one of my comments)

Other items could be postponed to later PRs:
- renaming the tables patient_list and patient_list_list
- White space problem

**Comment by jjgao** [comment]
I am going to merge and close this so that we can release by Wednesday. @adamabeshouse please log @pieterlukasse's points in another issue.

**Comment by jjgao** [comment]
@adamabeshouse you can create another pull if you have anything for this release.

**Comment by adamabeshouse** [comment]
#655 for some of the well-needed clarity changes requested by @pieterlukasse

**Pull request merged**